src/NATS.Client.Core/NatsConnection.cs

```diff
@@ -133,6 +133,9 @@ public NatsConnectionState ConnectionState
 
     internal ObjectPool ObjectPool => _pool;
 
+    // only used for internal testing
+    internal ISocketConnection TestSocket => _socket;
```

Comment:
Is this the best way to expose a private variable as Internal, but for Testing Only?  Use a `Test` prefix and a comment?  Or is there a better way - like a conditional compile flag for tests?